include/alpaka/core/BoostPredef.hpp

```diff
@@ -12,26 +13,24 @@
 #endif
 
 //---------------------------------------HIP-----------------------------------
-// __HIPCC__ is defined by hipcc (if either __CUDACC__ is defined)
+// __HIPCC__ is defined by hipcc when either __CUDACC__, __HIP__ or __HCC__ are defined.
 // https://github.com/ROCm-Developer-Tools/HIP/blob/master/docs/markdown/hip_porting_guide.md#compiler-defines-summary
 #if !defined(BOOST_LANG_HIP)
-#    if defined(__HIPCC__) && (defined(__CUDACC__) || defined(__HIP__))
-#        include <hip/hip_runtime.h>
-// HIP defines "abort()" as "{asm("trap;");}", which breaks some kernels
-#        undef abort
+#    if defined(__HIPCC__)
```

Comment:
Not sure if you can remove `__HIP__` because you can compile a pure C++ file with hipcc only if you set `-x hip` it will be use th e language HIP.

IMO we should not assume only because __HIPCC__ is set what is a marker for the compiler HIP is active.
Additional if we would like to use vanilla clang `__HIPCC__` is not set. We do not test currently vanilla clang but should not remove support for it.

Reply:
Could be that the current CI issue is triggered by this change

```
[ 57%] Building CXX object test/unit/acc/CMakeFiles/accTest.dir/src/AccNameTest.cpp.o
cd /builds/hzdr/crp/alpaka/build/test/unit/acc && /opt/rocm/llvm/bin/clang++ -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED -DALPAKA_ACC_GPU_HIP_ENABLED -DALPAKA_BLOCK_SHARED_DYN_MEMBER_ALLOC_KIB=47 -DALPAKA_CI -DALPAKA_DEBUG=0 -DALPAKA_OFFLOAD_MAX_BLOCK_SIZE="" -DBOOST_ALL_NO_LIB -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/builds/hzdr/crp/alpaka/include -isystem /builds/hzdr/crp/alpaka/thirdParty/catch2/src/catch2/.. -isystem /builds/hzdr/crp/alpaka/build/thirdParty/catch2/generated-includes -isystem /opt/boost/1.75.0/include -isystem /opt/rocm-5.0.0/hip/include -isystem /opt/rocm/include -isystem /opt/rocm-5.0.0/hip/../include -isystem /opt/rocm/llvm/lib/clang/14.0.0/include/.. -isystem /opt/rocm/rocrand/include -isystem /opt/rocm/hiprand/include -O3 -DNDEBUG -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-global-constructors -Wno-padded -Wno-extra-semi-stmt -Wno-unused-command-line-argument "-Wno-error=#warnings" -std=c++17 -pthread -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false -x hip --offload-arch=gfx906 -std=gnu++17 -MD -MT test/unit/acc/CMakeFiles/accTest.dir/src/AccNameTest.cpp.o -MF CMakeFiles/accTest.dir/src/AccNameTest.cpp.o.d -o CMakeFiles/accTest.dir/src/AccNameTest.cpp.o -c /builds/hzdr/crp/alpaka/test/unit/acc/src/AccNameTest.cpp
/builds/hzdr/crp/alpaka/test/unit/atomic/src/AtomicTest.cpp:25:12: error: no matching function for call to 'floatEqualExactNoWarning'
    return alpaka::math::floatEqualExactNoWarning(a, b);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builds/hzdr/crp/alpaka/test/unit/atomic/src/AtomicTest.cpp:30:12: error: no matching function for call to 'floatEqualExactNoWarning'
    return alpaka::math::floatEqualExactNoWarning(a, b);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```